### PR TITLE
feat(feedback): replace inline panel with milestone-triggered FeedbackDialog (#900)

### LINKED ISSUE #900: Feedback control gets too quiet after 2 views — testers can't find/recognize it
## Problem

Testers and organic users find it difficult to locate or recognize the feedback control on the results page. The previous inline `FeedbackPanel` rendered below the score card and demoted to a quiet, unbordered `-mt-8` star strip after 2 mounts (`ocv_feedback_seen >= 2`). This resulted in low organic feedback conversion because:
1. Asking for feedback immediately on mount (1st/2nd visit) is premature before the user has reviewed or exported their resume.
2. The compact inline strip lacked clear interactive affordances and was frequently missed or mistaken for static footer metadata.
3. Delighted users had no clear pathway to support the project (e.g. starring the GitHub repo), while users with praise or constructive critiques had no tailored prompt.

## Solution

Replace the inline `FeedbackPanel` with a **Multi-Step Feedback Interstitial (`FeedbackDialog`)** triggered at key value milestones and accessible via an ambient trigger button.

### 1. Trigger Pathways & Lifecycle
- **Automatic Milestone Trigger**: Automatically opens once after the user's **1st successful export** (PDF or Markdown via `ExportDialog` / `useDownloadPdf`).
- **Ambient Trigger**: An accessible `[★ Feedback]` button in `ParsedHeader` (and/or footer) that allows users and testers to open the feedback interstitial on demand at any time.
- **Persistence & Capping**:
  - `ocv_feedback_submitted`: Once submitted, the automatic prompt is suppressed.
  - `ocv_feedback_seen`: Tracks views; dismissed prompt is snoozed.
  - Suppressed entirely when analytics are disabled (`ANALYTICS_ENABLED === false`).

#### **Step 1: Quick Star Rating (1–5 ★)**
- Prominent 5-star rating control (`StarRating` primitive with 44×44px touch targets).
- Title: *"How did your resume turn out?"*
- Selecting a rating immediately transitions to Step 2 with sentiment-tailored routing.

#### **Step 2A: Positive Sentiment (4–5 ★) — "Delight & GitHub Star"**
- Title: *"Thank you! We're glad it helped."*
- Explains the open-source mission: *"OfflineCV is free, private, and runs in your browser. Starring our repo helps other job seekers discover it."*
- **GitHub Star CTA**: Embedded `GitHubStarCta` showing live star count with a direct link to the GitHub repository.
- **Optional Written Feedback**: Textarea for praise, favorite features, or new feature ideas (*"Any thoughts on what worked well or what to add next?"*).
- **Opt-in Updates Checkbox**: *"Keep me posted on major updates & new features"* (reveals optional email field).
- Actions: `[⭐ Star on GitHub]`, `[Submit & Close]`, `[Close]`.

#### **Step 2B: Constructive Sentiment (1–3 ★) — "Help Us Improve"**
- Title: *"How can we make OfflineCV better?"*
- **Category Pills**: `[Parsing]`, `[Scoring]`, `[UI / Editor]`, `[Export]`, `[Other]`.
- **Written Feedback Textarea**: *"Tell us what went wrong or what you'd change..."*.
- **Opt-in Follow-up Checkbox**: *"I'd like the team to follow up on this issue"* (reveals optional email field).
- Actions: `[Submit Feedback]`, `[Cancel / Skip]`.

### 3. Privacy & Design System Constraints
- Built using `@design-system` primitives (`Dialog`, `Button`, `StarRating`, `GitHubStarCta`, `Checkbox`).
- Zero PII collected unless the user explicitly opts in with an email address (`buildFeedbackProps`).
- In-place confirmations; no external toast/snackbar components.

## Acceptance Criteria

- [ ] Inline `FeedbackPanel` below the score card is removed; replaced by `FeedbackDialog` built on `@design-system`'s `Dialog` primitive.
- [ ] `FeedbackDialog` opens automatically upon the 1st completed PDF or Markdown export.
- [ ] An ambient `[★ Feedback]` button is added to `ParsedHeader` that opens `FeedbackDialog` on demand.
- [ ] Step 1 presents a 1–5 star rating with accessible labels and keyboard/touch navigation.
- [ ] Rating 4 or 5 stars routes to Step 2A (GitHub Star CTA + optional written praise/suggestions + optional update opt-in).
- [ ] Rating 1, 2, or 3 stars routes to Step 2B (Category pills + written issue description + optional contact opt-in).
- [ ] Submitting feedback sends sanitized props via `trackFeedback` in `src/lib/analytics.ts` and persists `ocv_feedback_submitted = "1"`.
- [ ] When `ANALYTICS_ENABLED` is false, all feedback tracking and prompts are cleanly suppressed.
- [ ] Existing unit tests and new flow tests for `FeedbackDialog` pass cleanly.

### PR DESCRIPTION
**Pull request by s-annam**
## Summary

Replaces the inline `FeedbackPanel` (a quiet star strip below the score card that got even quieter after 2 views) with a `FeedbackDialog` multi-step interstitial on `@design-system`'s `Dialog` primitive.

- **Automatic trigger**: opens once after the user's 1st completed PDF or Markdown export (not the audit report), deferred until the `ExportDialog` itself closes so it never stacks on top of the export-findings advisory.
- **Ambient trigger**: a `[★ Feedback]` button in `ParsedHeader`, available on demand.
- **Flow**: Step 1 is a 1–5 star rating; 4–5★ routes to a GitHub-star CTA + optional praise; 1–3★ routes to category pills + an issue textarea. Both paths end in an in-place, `aria-live` confirmation step rather than the dialog just vanishing.
- Capped via a new `ocv_feedback_dialog_seen` / `ocv_feedback_submitted` localStorage pair, cleanly suppressed when `ANALYTICS_ENABLED === false`.

Closes #900

## Adversarial review

Two rounds, pre-push, on the full branch diff.

**Round 1 — 4 blocking findings, all fixed:**
1. Auto-trigger reused the retired panel's `ocv_feedback_seen` key with different increment semantics — every existing browser already had it non-zero, so the new trigger would never fire. Fixed with a dedicated `ocv_feedback_dialog_seen` key.
2. The dialog could open on top of the still-open `ExportDialog`, burying the export-findings advisory. Fixed: the open is deferred to `ExportDialog`'s `onClose`.
3. `README.md`'s storage-key table still described the retired panel and a dead key (`ocv_star_cta_seen`). Reworded/removed to match current code.
4. Submitting closed the dialog with no confirmation (a11y regression vs `main`, which had a focused thank-you region). Fixed: an `aria-live="polite"` confirmation step with explicit Close.

**Round 2 — clean: true.** Each fix re-verified independently (not from the fix report) by tracing every caller and running the tests live. 5 nits surfaced and are left for a human pass, not blocking: a force-close edge case can leave the auto-trigger armed until the next export dialog interaction; `useGitHubStars()` double-fetches on cold cache; the step bodies lost `<form>`/Enter-to-submit and a visible email `<label>` vs the retired panel; the positive step's title thanks the user before anything is actually submitted.

## Validation

`npm run typecheck`, `npm run lint`, and the affected vitest suites (`useFeedbackDialog`, `FeedbackDialog`, `ExportDialog`, `ParsedHeader`, `Result`, `App.*`) all pass. `npm run check:fixtures` clean (no fixtures touched). Pre-push `verify` passed (fallow complexity/duplication findings are report-only per repo convention).

### CODE REVIEW

File: src/components/features/FeedbackDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    } catch {
+      // Best-effort: capture() is fire-and-forget; still confirm below.
+    }
+    onSubmitted();
+    setStep("thanks");
+  }
+
+  return (
+    <Dialog open={open} onClose={onClose} title={TITLES[step]} className="max-w-md">
+      {step === "rating" && (
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackDialog.tsx:98]: **Blocking — the auto-opened step has no way out.** This branch renders a `<p>` and a `StarRating` and nothing else; I mounted it to be sure: `STEP1 BUTTONS: []`, `STEP1 FOCUSABLES: ["INPUT:radio" x5]`. `Dialog` draws no close affordance of its own and sets no `closedby`, and native `<dialog>` + `showModal()` does not light-dismiss on a backdrop click — so Esc is the only exit, and touch devices have no Esc key. Since this is the one dialog in the tree that opens itself (after the first export), a phone user who does not want to rate has no dismissal path at all; step 2A/2B and `thanks` all give one, and so do `ExportDialog` (`Close`) and `ResumeChooserDialog` (`Cancel`).

A ghost `Close` beside the stars, calling `onClose`, restores the convention. It also gives the "dismissed prompt is snoozed" capping in #900 §1 an actual dismiss gesture to hang on.

File: src/components/features/FeedbackDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  // silently changes. Moving focus into the confirmation is also what makes a
+  // screen reader announce that the submission landed.
+  useEffect(() => {
+    if (step === "thanks") thanksRef.current?.focus();
+  }, [step]);
+
+  // No analytics sink → no dialog (a submission would silently vanish).
+  if (!ANALYTICS_ENABLED) return null;
+
+  function handleRate(value: number): void {
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackDialog.tsx:81]: **Blocking — routing on the first `onChange` locks keyboard users out of 4-5★.** `StarRating` is a native radio group, and its own docblock states the consequence: "arrow keys move between stars **and select** (native radio behaviour)". There is no way to move focus across radios *without* selecting, so for a user who tabs in, the first Arrow press selects star 2, `handleRate` fires, and the group unmounts into `FeedbackConstructiveStep`. They can never reach 4 or 5 — the positive branch, the GitHub CTA and the praise textarea are unreachable by keyboard — and with no Back on step 2, an overshoot is uncorrectable. A pointer mis-click hits the same wall.

This is also AC 4 of #900 ("a 1–5 star rating with accessible labels and keyboard/touch navigation") going unmet under a `Closes #900`.

Cheapest fix that keeps the no-extra-click design for pointer users: a `Back` button on both step-2 bodies that sets `step` back to `"rating"`. Gating the route behind an explicit Next would also work but costs the gesture the docblock is built around.

File: src/components/features/FeedbackDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+};
+
+/** What the step bodies collect — everything but the rating, which the
+ *  shell already knows from routing Step 1. */
+type StepSubmission = Omit<FeedbackArgs, "rating">;
+
+export function FeedbackDialog({ open, onClose, onSubmitted }: FeedbackDialogProps) {
+  const [step, setStep] = useState<Step>("rating");
+  const [rating, setRating] = useState(0);
+  const { count: starCount } = useGitHubStars();
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackDialog.tsx]: **Secondary — this fetches GitHub on every page load, for a dialog almost nobody opens.** `App.tsx:760` mounts `FeedbackDialog` unconditionally, `Dialog` renders its children even while closed, and `useGitHubStars` fires from a `[]`-dep effect. Mounted with `open: false` it still produces `FETCH CALLS: ["https://api.github.com/repos/offlinecv/OfflineCV"]`. `PageShell.tsx:108` already calls the same hook on every load, so on a cold cache the two race and spend two requests of the 60/hr unauthenticated per-IP budget on one number.

The PR body lists this as a known nit — the tidy fix is to call `useGitHubStars()` inside `FeedbackPositiveStep` instead, which only mounts on the 4-5★ branch. That deletes the `starCount` prop and the plumbing for it at the same time.

File: src/components/features/FeedbackDialog.test.tsx [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+  // primitive calls `showModal()` from an effect. Stubbed to a plain open so
+  // the tests exercise the dialog's CONTENT rather than the UA's modality —
+  // same stub `ExportDialog.test.tsx` uses.
+  HTMLDialogElement.prototype.showModal = function showModal(this: HTMLDialogElement) {
+    this.open = true;
+  };
+  HTMLDialogElement.prototype.close = function close(this: HTMLDialogElement) {
+    this.open = false;
+  };
+  vi.resetModules();
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackDialog.test.tsx:46]: **Secondary — this suite hits the network on every run.** `showModal`/`close` are stubbed and `analytics.ts` is mocked, but `useGitHubStars` is not, and `src/test-setup.ts`'s font shim delegates every non-font URL to the real `fetch` — so `api.github.com` is contacted for real, in CI included. `PageShell.test.tsx:91-96` stubs `fetch` for exactly this reason. It shows up in the wall clock too: 2106 ms here against 163-484 ms for the three neighbouring suites.

```suggestion
  // useGitHubStars fetches on mount and swallows a failure — stubbed so this
  // suite never touches the network (same as PageShell.test.tsx).
  vi.stubGlobal("fetch", vi.fn(() => Promise.reject(new Error("no network in tests"))));
  vi.resetModules();
```

File: src/components/features/FeedbackPositiveStep.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      />
+
+      <div className="flex flex-col gap-2">
+        <Checkbox
+          checked={wantsContact}
+          onChange={setWantsContact}
+          label="Keep me posted on major updates & new features"
+        />
+        {wantsContact && (
+          <input
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackPositiveStep.tsx]: **Secondary — these eleven lines are byte-identical to `FeedbackConstructiveStep.tsx:86-95`.** That is the exact pressure `Checkbox` was extracted under — its docblock says it "would otherwise have been the THIRD verbatim copy of the same six lines" — and the count is back to two before the feature has shipped.

The copy also drops the visible `<label>Your email</label>` the retired panel carried (`FeedbackPanel.tsx`, pre-deletion): a placeholder is not a label, so this is WCAG 3.3.2, and the placeholder vanishes the moment the user starts typing. One shared `EmailOptIn` (checkbox + revealed labelled field) owned by both steps closes both halves at once.

File: src/components/features/FeedbackConstructiveStep.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      />
+
+      <div className="flex flex-col gap-2">
+        <Checkbox
+          checked={wantsContact}
+          onChange={setWantsContact}
+          label="I'd like the team to follow up on this"
+        />
+        {wantsContact && (
+          <input
```
Comment (rohithgollapalli) [inline src/components/features/FeedbackConstructiveStep.tsx]: Second copy of the email opt-in field — see the note on `FeedbackPositiveStep.tsx:68`. Same eleven lines, same missing visible `<label>`.

**Review by rohithgollapalli — DISMISSED** [review]
Reviewed against #900's acceptance criteria first, then the diff, then the description. All nine ACs are implemented, the branch is one commit, and `typecheck` / `lint` / the four touched suites are green locally. Two findings block: both are in Step 1's rating gesture, and both hit keyboard and touch users specifically.

## Blocking

**1. Step 1 has no dismiss control, and this is the only dialog in the repo that opens itself.** `FeedbackDialog.tsx:105-118` renders a `<p>` and a `StarRating` — zero buttons. Verified by mounting it: `STEP1 BUTTONS: []`, `STEP1 FOCUSABLES: ["INPUT:radio" x5]`. The `Dialog` primitive draws no close affordance of its own, sets no `closedby`, and native `<dialog>`+`showModal()` does not light-dismiss on a backdrop click, so the only exit is Esc. On touch there is no Esc key, and this dialog opens **unprompted** after the first export. Every sibling dialog gives an explicit out — `ExportDialog.tsx:258` `Close`, `ResumeChooserDialog.tsx:137` `Cancel`; steps 2A/2B and `thanks` here do too. Only the auto-opened step does not.

**2. A keyboard-only user cannot reach a 4-5★ rating at all.** `StarRating` is a native radio group — its own docblock says "arrow keys move between stars **and select** (native radio behaviour)". `FeedbackDialog.tsx:88-91` routes on the first `onChange`, so Tab into the group + first Arrow press selects star 2 and immediately unmounts the group into `FeedbackConstructiveStep`. There is no way to move focus across radios *without* selecting, and no Back on step 2, so a keyboard or screen-reader user is confined to 1-2★ and cannot correct an overshoot. Mouse users hit the same wall on a mis-click. This is also AC 4 ("keyboard/touch navigation") not being met under a `Closes #900`.

Both fall out of one change: a ghost `Close` on the rating step, and a `Back` on steps 2A/2B that returns to `"rating"` (the routing-without-a-next-click stays intact for pointer users).

## Secondary

**3. `useGitHubStars()` runs at App mount for a dialog that is usually never opened** — `FeedbackDialog.tsx:65`. `App.tsx:760` mounts `FeedbackDialog` unconditionally, `Dialog` renders children even while closed, and the hook fetches from a `[]`-dep effect. Verified: mounted with `open: false`, `FETCH CALLS: ["https://api.github.com/repos/offlinecv/OfflineCV"]`. `PageShell.tsx:108` already calls the same hook on every page load, so on a cold cache both race and you spend two of the 60/hr unauthenticated budget for one number. The PR body lists this as a known nit; the clean fix is to move the hook into `FeedbackPositiveStep`, which only mounts on the 4-5★ branch — that also deletes the `starCount` prop threading.

**4. `FeedbackDialog.test.tsx` makes a real network request on every run.** It stubs `showModal`/`close` and mocks `analytics.ts`, but never `fetch` — and `src/test-setup.ts`'s font shim delegates every non-font URL to the real `fetch`, so the same `api.github.com` call above fires in CI. `PageShell.test.tsx:91-96` stubs it for exactly this reason. The suite's 2106 ms (vs 163-484 ms for its neighbours) is this.

**5. The email opt-in field is a verbatim duplicate across the two new step files** — `FeedbackPositiveStep.tsx:68-77` and `FeedbackConstructiveStep.tsx:86-95`, identical eleven lines. This is the pressure that produced `Checkbox` in the first place (its docblock: "would otherwise have been the THIRD verbatim copy of the same six lines"). It also drops the visible `<label>Your email</label>` the retired panel carried — placeholder plus `aria-label` only, which is WCAG 3.3.2. One `EmailOptIn` shared by both steps fixes both halves.

**6. `GitHubStarCta`'s `variant="card"` branch now has no consumers.** `FeedbackPanel`'s local `StarCtaOnce` was its only caller and went with the file; `grep -rn 'variant="card"' src/` returns nothing. Its docblock still describes it as "used inside FeedbackPanel's post-positive-feedback thank-you surface". Either delete the variant or point the docblock at the intended consumer. (`fallow` does not catch it — it is a prop value, not an export.)

## Nits

**7. Five docblocks still reference the deleted `FeedbackPanel`.** `Checkbox.tsx:16` is the one that is now actively wrong — "`FeedbackPanel` is still unmigrated ... it belongs in its own change" — this PR *is* that change, and both new steps use the primitive. Also `ReportGapSection.tsx:15`, `GitHubStarCta.tsx:13`, `ResumeLibrary.test.tsx:11`, `ReportGapSection.test.tsx:7-8`. None are in the diff, so no inline anchor.

**8. Nothing tests the App-level wiring end to end.** `useFeedbackDialog` and `FeedbackDialog` are each covered well, and `ExportDialog.test.tsx:437` pins `onResumeExported`, but no test asserts that export → close `ExportDialog` → `FeedbackDialog` opens through `App`.

**9. `GitHubStarCta`'s policy list says "✗ No star modal on app load"** and calls itself load-bearing. This PR still satisfies it (post-export, not app load), but it is the first time the CTA lives inside a self-opening modal — worth one line in that docblock so the next reader does not have to re-derive the distinction.

## Gates

| Gate | Result |
|---|---|
| `/code-review` correctness pass | folded in above |
| 3a fixture PII | skipped — no fixtures touched |
| 3b design-system / reuse | pass on the ESLint-enforced surface; finding 5 is the soft half |
| 3c style tokens | pass — the only hex-shaped matches are `#900`/`#421`/`#621` issue refs |
| 3d fallow | pass as a gate; 1 complexity finding on `App.tsx` (37 cyclomatic, pre-existing, +10 lines here), 16 clone groups (warn). Report-only per CLAUDE.md |
| 3e command-level | skipped — no skill/script/bash files in the diff |
| 3f description accuracy | **accurate**; it discloses findings 3 and the label half of 5 itself. Omits findings 1, 2, 4 and 6 |
| `typecheck` / `lint` | green |
| touched suites | 47 tests green (`FeedbackDialog`, `useFeedbackDialog`, `ExportDialog`, `ParsedHeader`) |

Verdict rule: ≥1 Blocking finding → `REQUEST_CHANGES`, and nothing was committed or pushed to the branch. It is already one commit, so the invariant is intact.

---
Reviewed by: Claude Opus 5 (high)

File: src/components/features/FeedbackDialog.tsx (hunk 1 above)
Comment (s-annam) [reply src/components/features/FeedbackDialog.tsx:98]: Fixed in `56493f0` — added a ghost `Close` button beside the stars on Step 1, matching the `ExportDialog`/`ResumeChooserDialog` convention.

File: src/components/features/FeedbackDialog.tsx (hunk 2 above)
Comment (s-annam) [reply src/components/features/FeedbackDialog.tsx:81]: Fixed in `56493f0` — added a `Back` button to both Step 2 bodies that returns to Step 1 without losing the rating, so an overshoot on the native radio group is recoverable.

File: src/components/features/FeedbackDialog.tsx (hunk 3 above)
Comment (s-annam) [reply src/components/features/FeedbackDialog.tsx]: Fixed in `56493f0` — `useGitHubStars()` now fires from inside `FeedbackPositiveStep` (only mounts on the 4-5★ branch) instead of unconditionally from `FeedbackDialog`. Dropped the `starCount` prop and its plumbing.

File: src/components/features/FeedbackDialog.test.tsx (hunk 4 above)
Comment (s-annam) [reply src/components/features/FeedbackDialog.test.tsx:46]: Fixed in `56493f0` — stubbed `fetch` to reject in `FeedbackDialog.test.tsx`'s `mountDialog` setup, same pattern as `PageShell.test.tsx`. Suite now runs in ~190-500ms instead of ~2.1s.

File: src/components/features/FeedbackPositiveStep.tsx (hunk 5 above)
Comment (s-annam) [reply src/components/features/FeedbackPositiveStep.tsx]: Fixed in `56493f0` — extracted a shared `EmailOptIn` (`src/design-system/shared/EmailOptIn.tsx`) used by both step bodies, with a real visible `<label>Your email</label>` instead of a placeholder-only field.

File: src/components/features/FeedbackConstructiveStep.tsx (hunk 6 above)
Comment (s-annam) [reply src/components/features/FeedbackConstructiveStep.tsx]: Fixed in `56493f0` — this copy now uses the same shared `EmailOptIn` component; see the note on `FeedbackPositiveStep.tsx`.

**Comment by s-annam** [comment]
Addressed all 6 review threads and collapsed to one commit (`56493f0`):

- **Step 1 had no dismiss path** — added a ghost `Close` beside the stars, matching `ExportDialog`/`ResumeChooserDialog`.
- **Keyboard lockout on 4-5★/1-3★ routing** — added a `Back` button to both step-2 bodies; the star group is a native radio, so a keyboard overshoot was previously unrecoverable.
- **`useGitHubStars` fired on every page load** — moved the fetch into `FeedbackPositiveStep`, which only mounts on the 4-5★ branch.
- **Test suite hit the network** — stubbed `fetch` in `FeedbackDialog.test.tsx`; suite now runs in ~190ms vs ~2.1s.
- **Duplicate email opt-in, no visible label** — extracted a shared `EmailOptIn` (`src/design-system/shared/EmailOptIn.tsx`) used by both step bodies, with a real `<label>` instead of placeholder-only.

`typecheck`/`lint` clean, full suite green (380 files / 6328 tests). Re-requesting review — ready for another look.

**Comment by s-annam** [comment]
Ready for the next review round.

### Summary of Latest Changes
1. **Unconditional Feedback Availability (#900)**:
   - Removed `ANALYTICS_ENABLED` gating from `FeedbackDialog`, `useFeedbackDialog`, `ParsedHeader`, `Result`, and `App`. Feedback triggers are no longer suppressed when `VITE_POSTHOG_KEY` is absent (local dev / OSS builds), allowing users to submit feedback at all times.
2. **Home Page & Pre-drop Feedback Trigger**:
   - Added an ambient `Feedback` link in the shared `PageShell` footer so users can submit feedback from the home page or any surface.
3. **Score Card Ambient Trigger**:
   - The `[★ Feedback]` button in `ParsedHeader` is unconditionally passed down and visible in the score card header row when parsed.
4. **Review Findings Addressed**:
   - Step 1 explicit dismiss control (`Close`).
   - Step 2 `Back` navigation to allow correcting ratings via keyboard/mouse.
   - Lazy-loading GitHub star fetching inside `FeedbackPositiveStep`.
   - Shared `EmailOptIn` component with visible label.
   - Removed obsolete `FeedbackPanel` references and cleaned up test network mocks.
5. **Verification**:
   - `npm run typecheck`, `npm run lint`, and all 380 vitest test suites (6,328 tests) pass.
   - Pre-push verification hook (`npm run verify`, `vite build`, fallow audit) passed cleanly.
   - Branch collapsed to a single commit (`61314d1`).

**Comment by s-annam** [comment]
Merging this now rather than holding it — I'm going to be away from the computer and don't want it sitting open while I'm offline.

Functionality is verified through the UI.

@rohithgollapalli — I'm merging ahead of clearing your review comments from 8/26, so they're not lost, just not blocking. Happy to take anything still outstanding as a follow-up.

**Review by offlinecv-approver — APPROVED** [review]
🕹️✔️
